packages/next/build/webpack-config.ts

```diff
@@ -738,12 +739,14 @@ export default async function getBaseWebpackConfig(
     if (baseRes !== res) {
       return
     }
+    const isNextExternal = res.match(/next[/\\]dist[/\\]next-server[/\\]/)
```

Comment:
```suggestion
    if (res.match(/next[/\\]dist[/\\]next-server[/\\]/)) {
      return `commonjs ${request}`
    }
```

I think this could be simplified to directly early exit here

Reply:
Further above this regex is used `/next[/\\]dist[/\\]next-server[/\\](?!lib[/\\]router[/\\]router)/`. Does this make sense here to or is that unrelated?

Reply:
Yeah looks like we can just return right away if this matches here and we should exclude `lib/router/router` here too since that relies on `process.env` values being injected by webpack currently